### ISSUE TITLE
Import of @vite-plugin-checker-runtime does not respect base option in Full URL format

### ISSUE THREAD

**Issue by Alexis2004** [fi3ework/vite-plugin-checker#246]
### Describe the bug

In our application, we host the html page and the JS code of our application on different domains. For example, the page is loaded from the https://landing-domain.local/ domain, and the application is loaded from the https://app-domain.local/ domain.

In the `vite.config.ts` we set the absolute base URL:
```
export default defineConfig({
  base: 'https://app-domain.local/', // Simplified example to show problem in dev environment
});
```

We are trying to include `vite-plugin-checker` as recommended by the documentation https://vite-plugin-checker.netlify.app/faq/integration.html#with-traditional-backend-require-version-0-5-4:
```
<script type="module" src="https://app-domain.local/@vite-plugin-checker-runtime-entry"></script>
```

The library is initialized correctly, an overlay appears in case of an error, but links to open a file with an error in the editor do not work.

The fact is that the generated links look like this: https://landing-domain.local/__open-in-editor?file=XXXXX.
That is, they use domain https://landing-domain.local/ instead of domain https://app-domain.local/.

If we look at the contents of the dynamically generated file https://app-domain.local/@vite-plugin-checker-runtime-entry, we will see there:
```
import { inject } from "/@vite-plugin-checker-runtime";
inject({
  overlayConfig: {},
  base: "/",
});
```

For some reason, the `base` parameter swallowed the domain name and left only the part related to the path (for example, if you set the base url as `https://app-domain.local/foo/bar`, then only `/foo/bar` will remain in this script).

### Reproduction

See above

### Expected behavior

Base url support in Full URL format will work.

The https://app-domain.local/@vite-plugin-checker-runtime-entry file will have the following content:
```
import { inject } from "/@vite-plugin-checker-runtime";
inject({
  overlayConfig: {},
  base: "https://app-domain.local/",
});
```

### System Info

```shell
System:
    OS: macOS 13.3.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 893.16 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.20.0 - /opt/homebrew/opt/node@16/bin/node
    Yarn: 3.1.1 - /opt/homebrew/bin/yarn
    npm: 8.19.4 - /opt/homebrew/opt/node@16/bin/npm
    Watchman: 2023.05.22.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 114.0.5735.133
    Edge: 114.0.1823.51
    Firefox: 114.0.1
    Safari: 16.4
```

### Validations

- [X] Read the [docs](https://github.com/fi3ework/vite-plugin-checker#readme).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by fi3ework** [comment]
This could be confirmed by testing two localhost domains. I'm wondering how you're using absolute URL as base. If we set an absolute base in vite config in dev mode. The resolved base will still be `/` so that's why this issue happens (seems like vite does not allow absolute URL in dev mode).

https://github.com/fi3ework/vite-plugin-checker/blob/e063617d68b40c6a3954ec477f104e00e2dde251/packages/runtime/src/components/Diagnostic.ce.vue#L24

Could you provide more details, such as did you using with a proxy and you could try using `vite --debug` to provide more debug info. Thanks!

**Comment by Alexis2004** [comment]
I researched the problem in more detail, and yes, you seem to be right - the absolute URL does not work as expected in dev mode and is only used for production builds. In my case, in dev mode, all assets are loaded correctly due to the fact that the correct domain is set in the `server.origin` option in `vite.config.ts` (the vite documentation mentions this option just in the context of setting up the dev environment https://vitejs.dev/config/server-options.html#server-origin).

For ease of reproducing the problem, I prepared 2 repositories with reproduction.

A simple NodeJS server running on port 3000 and loading an application from another host as an entry point:
```
cd <some-temp-path>
git clone git@github.com:Alexis2004/simple-http-server.git
cd simple-http-server
yarn install
yarn build
yarn dev
```

A simple Vue JS application with a dev server available on port 4000 that has a `vite-plugin-checker` installed and has intentionally introduced ESLint warning to display an overlay. Here I can set either an absolute URL in the `base` parameter or leave it empty, in either case the behaviour is the same:
```
cd <some-temp-path>
git clone git@github.com:Alexis2004/vite-plugin-checker-app-on-different-domain.git
cd vite-plugin-checker-app-on-different-domain
yarn dev
```

After launching the applications, just go to http://localhost:3000/ in your browser.

——

As a fix for the problem described above, I suggest using the value of the `server.origin` option as the plugin’s base URL to build the source code editor’s links in dev mode, if it is set. And in all other cases, the value of the `base` option that is currently used.
